src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/RestHubLifetimeManagerFactory.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.SignalR.Management
+{
+    internal class RestHubLifetimeManagerFactory : IServiceHubLifetimeManagerFactory
+    {
+        private readonly ServiceManagerContext _context;
+
+        public RestHubLifetimeManagerFactory(IOptions<ServiceManagerContext> context)
+        {
+            _context = context.Value;
+        }
+
+        public Task<IServiceHubLifetimeManager> Create(string hubName, CancellationToken cancellationToken)
+        {
+            var manager = new RestHubLifetimeManager(hubName, _context.ServiceEndpoints.Single(), _context.ProductInfo, _context.ApplicationName);
+            return Task.FromResult((IServiceHubLifetimeManager)manager);
```

Comment:
is the convert required?

Reply:
Yes, the compiler can not convert Task<RestHubLifetimeManager> to Task<IServiceHubLifetimeManager> automatically.